java/java.source.base/src/org/netbeans/api/java/source/GeneratorUtilities.java

```diff
@@ -2212,8 +2209,7 @@ private static ScriptEngine engine(FileObject fo) {
         if (obj instanceof String) {
             synchronized (GeneratorUtilities.class) {
                 if (manager == null) {
-                    ClassLoader loader = Lookup.getDefault().lookup(ClassLoader.class);
-                    manager = new ScriptEngineManager(loader != null ? loader : Thread.currentThread().getContextClassLoader());
+                    manager = Scripting.createManager();
```

Comment:
Also, why is the loader logic ignored now?

Reply:
The classloader logic is handled centrally in `Scripting.createManager()`.